server.js

```diff
@@ -0,0 +1,157 @@
+
+    // Prevents browser from caching and storing page
+    app.use(helmet.noCache());
+
+    // Allow loading resources only from white-listed domains
+    app.use(helmet.contentSecurityPolicy()); //csp deprecated
+
+    // Allow communication only on HTTPS
+    app.use(helmet.hsts());
+
+    // TODO: Add another vuln: https://github.com/helmetjs/helmet/issues/26
+    // Enable XSS filter in IE (On by default)
+    // app.use(helmet.iexss());
+    // Now it should be used in hit way, but the README alerts that could be
+    // dangerous, like specified in the issue.
+    // app.use(helmet.xssFilter({ setOnOldIE: true }));
+
+    // Forces browser to only use the Content-Type set in the response header instead of sniffing or guessing it
+    app.use(nosniff());
+    */
+
+    // Adding/ remove HTTP Headers for security
+    app.use(favicon(__dirname + "/app/assets/favicon.ico"));
+
+    // Express middleware to populate "req.body" so we can access POST variables
+    app.use(bodyParser.json());
+    app.use(bodyParser.urlencoded({
+        // Mandatory in Express v4
+        extended: false
+    }));
+
+    // Enable session management using express middleware
+    app.use(session({
+        // genid: (req) => {
+        //    return genuuid() // use UUIDs for session IDs
+        //},
+        secret: cookieSecret,
+        // Both mandatory in Express v4
+        saveUninitialized: true,
+        resave: true
+        /*
+        // Fix for A5 - Security MisConfig
+        // Use generic cookie name
+        key: "sessionId",
+        */
+
+        /*
+        // Fix for A3 - XSS
+        // TODO: Add "maxAge"
+        cookie: {
+            httpOnly: true
+            // Remember to start an HTTPS server to get this working
+            // secure: true
+        }
+        */
+
+    }));
+
+    /*
+    // Fix for A8 - CSRF
+    // Enable Express csrf protection
+    app.use(csrf());
+    // Make csrf token available in templates
+    app.use((req, res, next) => {
+        res.locals.csrftoken = req.csrfToken();
+        next();
+    });
+    */
+
+    // Register templating engine
+    app.engine(".html", consolidate.swig);
+    app.set("view engine", "html");
+    app.set("views", `${__dirname}/app/views`);
+    // Fix for A5 - Security MisConfig
+    // TODO: make sure assets are declared before app.use(session())
+    app.use(express.static(`${__dirname}/app/assets`));
+
+
+    // Initializing marked library
+    // Fix for A9 - Insecure Dependencies
+    marked.setOptions({
+        sanitize: true
+    });
+    app.locals.marked = marked;
+
+    // Application routes
+    routes(app, db);
+
+    // Template system setup
+    swig.setDefaults({
+        // Autoescape disabled
+        autoescape: false
+        /*
+        // Fix for A3 - XSS, enable auto escaping
+        autoescape: true // default value
+        */
+    });
+
+    // Insecure HTTP connection
+    http.createServer(app).listen(port, () => {
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- f30d523ff0c99b54928b336c897b6c21 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Cleartext Transmission of Sensitive Information
_Low CWE-319_
A web server is created using the insecure `HTTP` scheme. As a result, application data is transmitted over an insecure channel where it can be read and modified by attackers.



### How to fix
Potential mitigations include the following:
  1. Encrypt the data with a reliable encryption scheme before transmitting.
  2. When using web applications with SSL, use SSL for the entire session from login to logout, not just for the initial login page.
  3. Use tools and techniques that require manual (human) analysis, such as penetration testing, threat modeling, and interactive tools that allow the tester to record and modify an active session. These may be more effective than strictly automated techniques. This is especially the case with weaknesses that are related to design and business rules.
  4. Configure servers to use encrypted channels for communication, which may include SSL or other secure protocols.